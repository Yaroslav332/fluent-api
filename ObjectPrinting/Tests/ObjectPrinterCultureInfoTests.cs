using System.Globalization;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace ObjectPrinting.Tests
{
    [UseReporter(typeof(DiffReporter))]
    [TestFixture]
    public class ObjectPrinterCultureInfoTests
    {
        [Test]
        public void SetCulture_Should_ChangeFormat()
        {
            var p = new Person {Name = "Alex", Height = 35.5, Weight = 83.3};
            var cultureInfo = new CultureInfo("ru");
            var printer = ObjectPrinter.For<Person>()
                .SetCulture(cultureInfo);
            Approvals.Verify(printer.PrintToString(p));
        }

        [Test]
        public void SetCulture_Should_OverridePreviousSetCulture()
        {
            var p = new Person {Name = "Alex", Height = 35.5, Weight = 83.3};
            var cultureInfo1 = new CultureInfo("ru");
            var cultureInfo2 = new CultureInfo("en");
            var printer = ObjectPrinter.For<Person>()
                .SetCulture(cultureInfo1)
                .SetCulture(cultureInfo2);
            Approvals.Verify(printer.PrintToString(p));
        }

        [Test]
        public void SetCulture_ShouldNot_OverridePreviousMethod()
        {
            var p = new Person {Name = "Alex", Height = 35.5, Weight = 83.3};
            var cultureInfo = new CultureInfo("ru");
            var printer = ObjectPrinter.For<Person>()
                .ConfigForProperty(p => p.Height)
                .UseSerializeMethod(f => "123haha")
                .SetCulture(cultureInfo);
            Approvals.Verify(printer.PrintToString(p));
        }

        [Test]
        public void UseSerializeMethod_Should_OverridePreviousSetCulture()
        {
            var p = new Person {Name = "Alex", Height = 35.5, Weight = 83.3};
            var cultureInfo = new CultureInfo("ru");
            var printer = ObjectPrinter.For<Person>()
                .SetCulture(cultureInfo)
                .ConfigForProperty(p => p.Height)
                .UseSerializeMethod(f => "123haha");
            Approvals.Verify(printer.PrintToString(p));
        }
    }
}