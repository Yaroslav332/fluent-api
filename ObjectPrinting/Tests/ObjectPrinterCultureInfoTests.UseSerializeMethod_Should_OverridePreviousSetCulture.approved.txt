Person
	Id = 00000000-0000-0000-0000-000000000000
	Name = Alex
	Height = 123haha
	Weight = 83,3
	Birthday = 01.01.0001 0:00:00
	RunRatio = 0
	Age = 0
	RunSpeed = 0
	Parent = null
